hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImpl.java

```diff
@@ -233,11 +233,12 @@ private void applyInitialState(int version, Collection<MemberInfo> memberInfos)
     private MemberListSnapshot createSnapshot(int memberListVersion, Collection<MemberInfo> memberInfos) {
         LinkedHashMap<UUID, Member> newMembers = new LinkedHashMap<>();
         for (MemberInfo memberInfo : memberInfos) {
-            MemberImpl member = new MemberImpl.Builder(memberInfo.getAddress()).version(memberInfo.getVersion())
+            MemberImpl member = new MemberImpl.Builder(memberInfo.getAddressMap())
```

Comment:
When this memberInfo is coming from 4.0 cluster, it will be null. The constructor will throw NullPointerException. 
I prefer not to put `buildForClientView` to `MemberImpl.Builder` for the client. The client logic should be handled in client packages. 
Can we check if the addressMap is null, and create the builder with `address` if it is there. 
Otherwise, 



Can we also put a backward compatibility test against 4.1 client against 4.0 cluster to make sure ?

Reply:
Hello @sancar , thank you for the review.

> When this memberInfo is coming from 4.0 cluster, it will be null. The constructor will throw NullPointerException.

good catch, fixed

> I prefer not to put buildForClientView to MemberImpl.Builder for the client. The client logic should be handled in client packages.

Agreed, refactored.

> Can we also put a backward compatibility test with 4.1 client against 4.0 cluster to make sure ?

Can you please give me any clue on how to do that?